dbms/src/DataStreams/ConstantsBlockInputStream.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/Exception.h>
+#include <DataStreams/IBlockInputStream.h>
+
+namespace DB
+{
+class ConstantsBlockInputStream : public IBlockInputStream
+{
+public:
+    ConstantsBlockInputStream(const Block & header, UInt64 rows_, UInt64 max_block_size_)
+        : header(header)
+        , remaining_rows(rows_)
+        , max_block_size(std::max(1, max_block_size_))
+    {
+        for (const auto & col : header)
+        {
+            RUNTIME_CHECK(col.column != nullptr && col.column->isColumnConst());
+        }
+        if unlikely (header.columns() == 0)
```

Comment:
I prefer to throw error there because just set `remaining_rows` to 0 will return different result as the spill input. In fact, I think we can need disable spill if the spill input schema is empty block.

Reply:
Ok, I feel that an error can be reported directly here. An empty header is abnormal even if it is not a spill.